crates/sui-core/src/storage.rs

```diff
@@ -338,3 +340,151 @@ impl WriteStore for RocksDbStore {
         Ok(())
     }
 }
+
+pub struct RestReadStore {
+    state: Arc<AuthorityState>,
+    rocks: RocksDbStore,
+}
+
+impl RestReadStore {
+    pub fn new(state: Arc<AuthorityState>, rocks: RocksDbStore) -> Self {
+        Self { state, rocks }
+    }
+}
+
+impl ObjectStore for RestReadStore {
+    fn get_object(
+        &self,
+        object_id: &sui_types::base_types::ObjectID,
+    ) -> sui_types::storage::error::Result<Option<Object>> {
```

Comment:
nit: replace the fully qualified path (and all the rest below).

Reply:
+1 on this -- maybe not for `Result` if there are multiple kinds of result floating around, but I 
 can't see the argument for keeping `ObjectID`, `VersionNumber` fully-qualified here.